### PR TITLE
fix(label): convert rtl label

### PR DESCRIPTION
**Pull request by hugh-qantas**
## Description

migrate label to use React Testing Library pattern

## Related issues

https://app.shortcut.com/qantasaccommodation/story/115272/convert-tests-to-rtl-and-rename-with-the-jsx-extension-for-label

---

**PR check-list**

🚨 **IMPORTANT:** Please review the [guidelines for contributing](./CONTRIBUTING.md) to this repository first! 👀

- [x] You've requested a review from at least ONE person. Preferably a roo-ui maintainer ([Philip](https://github.com/philipwindeyer), [Eric](https://github.com/eneo5541), [Mike](https://github.com/memcc), [Marcel](https://github.com/Anthropic), [Sam](https://github.com/SamuSan), [Long](https://github.com/KieraDOG)) or a Roo UI working-group human.
  - Note: If your PR has not been reviewed within two days of requesting a review, feel free to seek reviews from other [team members of Qantas Hotels](https://github.com/orgs/hooroo/people).
- [x] Your final merge commit message is in the format `<type>(<scope>): <description>` as per [the contributing guide](https://github.com/hooroo/roo-ui/blob/master/.github/CONTRIBUTING.md#commit-naming)
  - Example: `fix(dependencies): minor version bump to resolve security vulnerability in set-value`

### CODE REVIEW

File: src/components/Label/Label.test.jsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  );
+
+  const label = screen.getByTestId('my-label');
+
+  expect(label).toBeInTheDocument();
+  expect(label).not.toHaveStyleRule('height', '1px');
+  expect(label).not.toHaveStyleRule('position', 'absolute');
+});
+
+test('should hide label visually when given hidden prop', () => {
```
Comment (ajlu) [inline src/components/Label/Label.test.jsx:24]:
```suggestion
test('hides label visually when hidden is true', () => {
```
just to be more specific - `given hidden prop` could also apply for `hidden={false}`?

File: src/components/Label/Label.test.jsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  render(
+    <Box>
+      <Label id="username" data-testid="my-label">
+        Hello world
+      </Label>
+      <Input aria-labelledby="username" />
+    </Box>,
+  );
+
+  const label = screen.getByTestId('my-label');
```
[the remark this answers is not in the document]
Comment (hugh-qantas) [reply src/components/Label/Label.test.jsx]: have removed `Box` and `data-testid`

File: src/components/Label/Label.test.jsx (hunk 1 above)
Comment (hugh-qantas) [reply src/components/Label/Label.test.jsx]: have made test title more specific

**Comment by hugh-qantas** [comment]
@ajlu - I've made the changes as per PR comments. Could you please review?
Thanks

File: src/components/Label/Label.test.jsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      <Input id="username" />
+    </>,
+  );
+  const label = screen.getByText(/name/i);
+
+  expect(label).toBeInTheDocument();
+  expect(label).toHaveStyleRule('height', '1px');
+  expect(label).toHaveStyleRule('position', 'absolute');
+});
+
```
Comment (ajlu) [inline src/components/Label/Label.test.jsx:37]: Could you add another test like https://github.com/hooroo/roo-ui/blob/master/src/components/Flex/Flex.test.jsx#L20 to test the custom props supported by `<Label />`?

**Review by ajlu — APPROVED** [review]

File: src/components/Label/Label.test.jsx (hunk 3 above)
Comment (hugh-qantas) [reply src/components/Label/Label.test.jsx:37]: done. Added unit test for custom props supported by Label

**Review by klubiniecki-qantas — APPROVED** [review]
👏

**Review by michael-velsigne-QTW — APPROVED** [review]

**Pull request merged**